ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/KeyboardRepository.swift

```diff
@@ -11,21 +11,22 @@ import Foundation
 public protocol KeyboardRepository: class {
   typealias CompletionHandler = (Error?) -> Void
 
-  weak var delegate: KeyboardRepositoryDelegate? { get set }
+  var delegate: KeyboardRepositoryDelegate? { get set }
   var languages: [String: Language]? { get }
+  var keyboards: [String: Keyboard]? { get }
 
   func installableKeyboard(withID keyboardID: String, languageID: String) -> InstallableKeyboard?
   func fetch(completionHandler: CompletionHandler?)
 }
 
 public extension KeyboardRepository {
   public func installableKeyboard(withID keyboardID: String, languageID: String) -> InstallableKeyboard? {
-    guard let language = languages?[languageID] else {
-      return nil
-    }
-    guard let keyboard = language.keyboards?.first(where: { $0.id == keyboardID }) else {
+    guard let keyboard = keyboards?.first(where: { $0.key == keyboardID })?.value else {
       return nil
     }
+    
+    let language = keyboard.languages?.first(where: {$0.id == languageID}) ?? keyboard.languages?.first ?? languages?[languageID] ?? Language(name: languageID, id: languageID, keyboards: [keyboard], font: nil, oskFont: nil)
```

Comment:
Is it possible to split this over multiple lines and comment on what is happening? It's a bit of a complex clause and a bit side-effecty :)

Reply:
It is

Reply:
@mcdurdin I have split this into four lines and am adding appropriate comments. The question now is whether we still need/want the fourth part? It' pretty kludgy and now will hopefully be unneeded. (But it does offer the ultimate safety net if something has gone wrong - which should help to ensure that we never find out about it and fix it.)

Reply:
I've removed the kludgy fallback. It's at least not needed to solve this issue and it's probably a bad idea.